### DISCUSSION TITLE
how to add custom dropdown field in Elsa setting page UI and how to save this dropdown value in the workflow definition table?

### DISCUSSION THREAD

**Discussion by SatishKolte** [elsa-workflows/elsa-core#4009] [Q&A]
I need to Add Custom Drop down Field in this section. So How to do this

![image](https://user-images.githubusercontent.com/96614575/236757069-a4f8082d-5baa-428e-be79-8460212eda7b.png)

**Comment by SatishKolte** [comment]
@sfmskywalker 
Can you provide an answer as early as possible because I am stuck now, and your answer would be helpful for me?

**Reply by sfmskywalker** [reply]
Hi @SatishKolte 

Currently, that screen does not provide any extensibility points, but we can probably add it.
Can you describe your use case? This would help us get a better understanding of how we might make this UI extensible.

**Comment by SatishKolte** [comment]
Thanks for your insightful suggestion.

In relation to my use case, I am currently working on differentiating workflows, and therefore, I am in need of a custom field to be added to the workflow definition table.

To provide a clearer understanding, let me illustrate with an example from my project. Within the surgery and precert module, we have multiple workflows. Whenever I initiate the surgery workflow, it is crucial to simultaneously trigger all workflows that fall under the surgery type.

Considering the aforementioned requirement, it becomes imperative to introduce a new field that can accurately identify the specific type of workflow. This custom field would greatly enhance our ability to efficiently manage and differentiate workflows as per their respective types.

Thank you for taking the time to review my request. I eagerly await further discussions regarding the implementation options for this crucial addition.

Thank and Regards
 Satish
________________________________

**Reply by sfmskywalker** [reply]
I see. That makes sense! For Elsa 3, we are going to add a tagging feature that lets you add tags to a workflows, by which you can then query workflows as per your description.

Short term, I won't be able to add this to Elsa 2 due to other priorities. Of course, you're welcome to open an issue requesting this feature. With luck, someone from the community wants to pick it up, or perhaps you'd like to take a stab at it yourself and submit a PR 😄

**Comment by SatishKolte** [comment]
Hi @sfmskywalker 
I am currently encountering with the Auto Mapper and Elsa packages in my project.

Specifically, when attempting to open the Web Host Project "abp-zero-template" with .NET Framework 6.0 and version 11.1.0, I am encountering an error. This error seems to be related to the compatibility of the AutoMapper package.

**System.MissingMethodException: 'Method not found: 'Void AutoMapper.IMappingExpressionBase`3.ConvertUsing(System.Linq.Expressions.Expression`1<System.Func`2<!0,!1>>)'.'**

![b26110302db5de710c1f3a0305f317d4](https://github.com/elsa-workflows/elsa-core/assets/96614575/82abd3fe-6acc-4b48-83b5-2dade4c3582d)

Furthermore, I have noticed that the latest stable version of the Elsa package is 2.11.1. However, even after trying different versions, I am consistently encountering the same error.

I would greatly appreciate any guidance or assistance you can provide in resolving this issue.

Thanks

**Reply by sfmskywalker** [reply]
@SatishKolte It's better if you start a new discussion or issue to prevent multiple topics being discussed in a single discussion. Thank you 😄
